### ISSUE TITLE
RepresentationError raised when saving a new Configuration object that consists sequences

### ISSUE THREAD

**Issue by Rachidrh** [NetherlandsForensicInstitute/confidence#95]
The example below will raise a  `yaml.representer.RepresenterError:` exception.
`
import confidence
from confidence import Configuration

a = Configuration({'a_1': '1', 'a_2': [2, 2, 2]})
b = Configuration({'b_1': '2', 'b_2': [2, 2, 2]})

confidence.dumpf(Configuration({'a': a, 'b':b}), 'output.yaml')
`
The workaround is  use `confidence.dumpf(Configuration({'a': a._source, 'b':b._source}), 'output.yaml')`

**Comment by akaIDIOT** [comment]
You're right, there should be no need to use protected variables. The error that comes back is related to the lists, that show up to the YAML representer as a `ConfigurationSequence`. The real issue here is that creating a new `Configuration` should make sure there's no `Configuration` objects further down the line; the `dict` you're providing should be unwrapped / unraveled into simple `dict`s and `list`s, which would make the dumpers able to dump any kind of `Configuration` object again.

Thanks for submitting this, let's look into how we can best fix this :)

**Comment by akaIDIOT** [comment]
Calling this as part of the collect / merge stap during init of `Configuration` fixes the issue:

```python
def unwrap(source):
    while isinstance(source, Configuration):
        source = source._source

    if isinstance(source, Mapping):
        return {key: _unwrap(value) for key, value in source.items()}
    else:
        return source
```

Not sure yet if that's the best option, as that is a fairly expensive recursive cleaning of the source, just before it gets recursively traversed again to be merged. Ideally, this should be integrated into the merge, but that would currently cause a circular import :sweat_smile: 

It's fixable, just need to wedge it into the right spot :thinking:

**Comment by Rachidrh** [comment]
This suggested solutions looks good! I don't think that the (additional) recursion will cause much harm as in most cases a Configuration will not have deeply nested  mappings/sequences.

**Comment by akaIDIOT** [comment]
Whoops, this has been fixed by the merge.
